mmv1/third_party/terraform/services/bigquery/data_source_google_bigquery_datasets.go

```diff
@@ -0,0 +1,137 @@
+package bigquery
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleBigQueryDatasets() *schema.Resource {
+
+	dsSchema := map[string]*schema.Schema{
```

Comment:
Could you please implement a top-level field `filter` that will be used to set [the filter query parameter](https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/list#query-parameters)? That query parameter isn't available when listing tables, so I didn't mention it before, but we should support it for listing datasets.

Adding filtering will impact:
- The id of the data source - [see example here](https://github.com/BBBmau/magic-modules/blob/17a2e4db34706f557b7d864b3b20efe5a78381cf/mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_versions.go#L90-L93)
- The filter param will need to be included in the URL if filter is set - [example here
](https://github.com/BBBmau/magic-modules/blob/17a2e4db34706f557b7d864b3b20efe5a78381cf/mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_versions.go#L181-L193)

Reply:
Hi @SarahFrench! Thank you so much for the feedback. Per @wj-chen's comment here: https://github.com/GoogleCloudPlatform/magic-modules/pull/11552#pullrequestreview-2278675204
I'll be reverting these changes and spinning up an identical PR soon. I'll address this feedback over there 👍🏼.